plugin-niassa/src/main/java/ca/on/oicr/gsi/shesmu/niassa/WorkflowConfiguration.java

```diff
@@ -129,9 +230,11 @@ public String applyTuple(Stream<Field<Integer>> contents) {
   private long[] previousAccessions;
   private boolean relaunchFailedOnUpgrade;
   private List<String> services = Collections.emptyList();
-  private InputLimsKeyProvider type;
+  private InputLimsKeyProvider<WorkflowAction> type;
```

Comment:
I think it would be better to change the `InputLimsKeyProvider` to avoid having to duplicate this in all the `.niassawf` files. If you make it look like this:
```
public interface InputLimsKeyProvider {
  <T extends Action> CustomActionParameter<T> parameter();
}
```
Then the method will be generic without the interface being generic and the two can share an implementation.